### PR TITLE
Move resource XML attribute names into resource_info per Crudelios.

### PR DESCRIPTION
**Pull request by PrettyFlower**

### CODE REVIEW

**Comment by PrettyFlower** [comment]
I have no idea why this won't compile.

**Comment by crudelios** [comment]
You need to add `game/resource.c` to `res/asset_packer/CMakeLists.txt`

File: src/core/xml_parser.c [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #include "xml_parser.h"
 
 #include "core/log.h"
+#include "game/resource.h"
```
Comment (crudelios) [inline src/core/xml_parser.c:4]: Duplicate include, you're already including this in the header

File: src/core/xml_parser.c [hunk 2]
```diff
@@ -261,6 +262,21 @@ int xml_parser_get_attribute_enum(const char *key,
...
 
+resource_type xml_parser_get_resource(const char *key)
+{
+    const char *value = get_attribute_value(key);
+    if (!value) {
+        return RESOURCE_NONE;
+    }
+    for (resource_type i = RESOURCE_MIN; i < RESOURCE_MAX; i++) {
+        const char *resource_name = resource_get_data(i)->xml_attr_name;
+        if (resource_name != NULL && compare_multiple(resource_name, value)) {
```
Comment (crudelios) [inline src/core/xml_parser.c:273]: I don't think there is any case where the resource xml attr isn't set. Also, use `0`, not `NULL` please.

**Review by crudelios — CHANGES_REQUESTED** [review]
Some small changes needed

File: src/core/xml_parser.c (hunk 1 above)
Comment (PrettyFlower) [reply src/core/xml_parser.c:4]: I was weak and desperate. I just wanted the thing to compile :'(

**Pull request merged**